redis/src/main/scala/zio/redis/Interpreter.scala

```diff
@@ -60,35 +62,39 @@ trait Interpreter {
         Promise.make[RedisError, String].flatMap(p => pendingRequests.offer(Request(command, p)).as(p))
 
       private val executeNext: UIO[Any] =
-        pendingRequests.take.flatMap { request =>
+        pendingRequests.take.map { request =>
           val exchange =
-            IO {
-              unsafeSend(request.command)
-              unsafeReceive()
+            IO.effect {
+              try {
+                unsafeSend(request.command)
+                unsafeReceive()
+              } catch {
+                case t: Throwable => throw RedisError.ProtocolError(t.getMessage)
+              } finally readinessFlag.set(true)
             }
 
-          val completer = exchange.catchAll(RedisError.make).ensuring(UIO(readinessFlag.set(true)))
-
-          request.result.completeWith(completer)
+          request.result.unsafeDone(exchange.refineToOrDie[RedisError])
         }
 
       private def unsafeReceive(): String = {
-        val sb        = new StringBuilder
+        val cb        = ChunkBuilder.make[Byte]()
         var readBytes = 0
 
         // TODO: handle -1
         while (readBytes == 0) {
           channel.read(response)
           response.flip()
 
-          readBytes = response.limit()
+          readBytes = response.remaining()
 
-          sb ++= new String(Arrays.copyOf(response.array(), readBytes), UTF_8)
+          cb ++= Chunk.fromByteBuffer(response)
```

Comment:
I don't know if `Chunk` will copy the byte buffer, so this may not be safe. I think if you call `.materialize`, it will make an array copy. We should verify.

Reply:
It should be:

```scala
    val dest = Array.ofDim[Byte](buffer.remaining())
    val pos  = buffer.position()
    buffer.get(dest)
    buffer.position(pos)
    Chunk.fromArray(dest)
```

This would be problematic if buffer isn't flipped, otherwise, I don't think it'll make any problems.